File: src/jEvropa2/HtmlParser.java
package jEvropa2;

import jEvropa2.parser.ShowParser;
import jEvropa2.parser.ItemParser;
import jEvropa2.data.Item;
import jEvropa2.data.Show;
import org.jsoup.nodes.Element;
import org.jsoup.select.Elements;

import java.net.MalformedURLException;
import java.net.URL;
import java.util.LinkedHashSet;
import java.util.Set;

/**
 * Creates sets of {@link Show} and {@link Item} from html code.
 *
 * @author Roman Zelenik
 */
public class HtmlParser {

    ShowParser showParser;
    ItemParser itemParser;

    public HtmlParser() {
        this.showParser = new ShowParser();
        this.itemParser = new ItemParser();
    }

    public Set<Show> parseShows(Elements elements) {
        Set<Show> showsSet = new LinkedHashSet<>();
        Show newShow;
        for (Element element : elements) {
            try {
                newShow = showParser.parse(element);
                showsSet.add(newShow);
            } catch (MalformedURLException ex) {
                ex.printStackTrace();
            }
        }
        return showsSet;
    }

    public Set<Item> parseAudioShowItems(Elements elements) {
        Set<Item> itemsSet = new LinkedHashSet<>();
        Item newItem;
        for (Element element : elements) {
            try {
                newItem = itemParser.parseAudio(element);
                itemsSet.add(newItem);
            } catch (MalformedURLException ex) {
                ex.printStackTrace();
            }
        }
        return itemsSet;
    }

    public Set<Item> parseVideoShowItems(Elements elements) {
        Set<Item> itemsSet = new LinkedHashSet<>();
        Item newItem;
        for (Element element : elements) {
            try {
                newItem = itemParser.parseVideo(element);
                itemsSet.add(newItem);
            } catch (MalformedURLException ex) {
                ex.printStackTrace();
            }
        }
        return itemsSet;
    }

    /**
     *
     * @param element
     * @return Item without web site url
     */
    public Item parseActiveAudioShowItem(Element element) {
        Item item = null;
        try {
            item = itemParser.parseActiveAudio(element);
        } catch (MalformedURLException ex) {
            ex.printStackTrace();
        }
        return item;
    }

    public Item parseActiveVideoShowItem(Element element) {
        Item item = null;
        try {
            item = itemParser.parseActiveVideo(element);
        } catch (MalformedURLException ex) {
            ex.printStackTrace();
        }
        return item;
    }

    public URL parseMp3Url(Element script) {
        URL url = null;
        try {
            url = itemParser.parseMp3Url(script.html());
        } catch (MalformedURLException ex) {
            ex.printStackTrace();
        }
        return url;
    }

    public URL parseMp4Url(Element script) {
        URL url = null;
        try {
            url = itemParser.parseMp4Url(script.html());
        } catch (MalformedURLException ex) {
            ex.printStackTrace();
        }
        return url;
    }

    public URL parseNextPageUrl(Element element) {
        URL url = null;
        try {
            url = showParser.parseNextPage(element);
        } catch (MalformedURLException ex) {
            ex.printStackTrace();
        }
        return url;
    }
}

File: src/main/kt/cz/skup5/jEvropa2/HtmlParser.kt
package cz.skup5.jEvropa2

import cz.skup5.jEvropa2.data.Item
import cz.skup5.jEvropa2.data.Show
import cz.skup5.jEvropa2.parser.ItemParser
import cz.skup5.jEvropa2.parser.ShowParser
import org.jsoup.nodes.Element
import org.jsoup.select.Elements
import java.net.MalformedURLException
import java.net.URI
import java.util.*

/**
 * Creates sets of [Show] and [Item] from html code.
 *
 * @author Skup5
 */
class HtmlParser {

    private val showParser: ShowParser = ShowParser()
    private val itemParser: ItemParser = ItemParser()

    /**
     * Returns parsed set of [Show]s from the given [elements].
     */
    @Deprecated("It works, but will be removed.",
            ReplaceWith("ShowDao.get()", "cz.skup5.jEvropa2.dao.ShowDao"))
    fun parseShows(elements: Elements): Set<Show> {
        val showsSet = LinkedHashSet<Show>()

        for (element in elements) {
            showsSet += showParser.parse(element)
        }
        return showsSet
    }

    /**
     * Returns parsed set of audio [Item]s from the given [elements].
     */
    @Deprecated("It works, but will be removed.",
            ReplaceWith("ItemDao.get(show, page, perPage)", "cz.skup5.jEvropa2.dao.ItemDao"))
    fun parseAudioShowItems(elements: Elements): Set<Item> {
        val itemsSet = LinkedHashSet<Item>()

        for (element in elements) {
            itemsSet += itemParser.parseAudio(element)
        }
        return itemsSet
    }

    fun parseVideoShowItems(elements: Elements): Set<Item> {
        val itemsSet = LinkedHashSet<Item>()
        var newItem: Item
        for (element in elements) {
            try {
                newItem = itemParser.parseVideo(element)
                itemsSet.add(newItem)
            } catch (ex: MalformedURLException) {
                ex.printStackTrace()
            }

        }
        return itemsSet
    }

    /**
     * Returns parsed active audio [Item] from the given [element].
     */
    @Deprecated("It works, but will be removed.", ReplaceWith("ItemDao.get(show, page, perPage)", "cz.skup5.jEvropa2.dao.ItemDao"))
    fun parseActiveAudioShowItem(element: Element): Item = itemParser.parseActiveAudio(element)

    @Deprecated("It doesn't work. Will be repaired.", level = DeprecationLevel.ERROR)
    fun parseActiveVideoShowItem(element: Element): Item? {
        var item: Item? = null
        try {
            item = itemParser.parseActiveVideo(element)
        } catch (ex: MalformedURLException) {
            ex.printStackTrace()
        }

        return item
    }

    fun parseMp3Url(dataJSONElement: Element): URI {
        return itemParser.parseMediaUrl(dataJSONElement)
    }

    @Deprecated("It doesn't work. Will be repaired.", level = DeprecationLevel.ERROR)
    fun parseMp4Url(script: Element): URI {
        return itemParser.parseMp4Url(script.html())
    }

    @Deprecated("It doesn't work. Will be repaired.", level = DeprecationLevel.ERROR)
    fun parseNextPageUrl(element: Element): URI {
        return showParser.parseNextPage(element)
    }
}
